presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlock.java

```diff
@@ -334,7 +334,7 @@ protected void ensureHashTableLoaded()
 
             int[] offsets = getOffsets();
             int elementCount = getRawKeyBlock().getPositionCount();
-            int mapCount = getPositionCount();
+            int mapCount = offsets.length - 1;
```

Comment:
What about changing the variable name? eg `rawMapCount`? 

`mapCount` looks like "number of maps in the MapBlock" (which is what `positionCount` is all about). Here, it looks like this is "number of maps in the MapBlock, and all other MapBlocks based on same raw blocks"

Same for  `elementCount` in the line above.